### PR TITLE
Java appsync sam

### PR DESCRIPTION
**Pull request by khairulhabibataws**
*Issue #, if available:*

*Description of changes:*
This is to provide java version of appsync to dynamodb mock and integration test using codepipeline

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by rohanmeh** [comment]
closes #119

File: java-test-samples/java-appsync-sam/src/test/java/com/example/MockTest.java [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import software.amazon.awssdk.services.appsync.AppSyncClient;
+import software.amazon.awssdk.services.appsync.model.EvaluateMappingTemplateRequest;
+import software.amazon.awssdk.services.appsync.model.EvaluateMappingTemplateResponse;
+
+public class MockTest {
+
+    private static String locationContext = "{\"arguments\":\n    {\n        \"locationid\": \"1234567890\",\n        \"name\": \"Location Name\",\n        \"description\": \"Location Description\",\n        \"imageUrl\": \"https://www.example.com/image.jpg\"\n    },\n\"result\": {\n    \"locationid\": \"1234567890\",\n    \"imageUrl\": \"https://www.example.com/image.jpg\",\n    \"name\": \"Location Name\",\n    \"description\": \"Location Description\",\n    \"timestamp\": \"2023-01-01T00:00:00.000Z\"\n}}";
```
Comment (dmahapatro) [inline java-test-samples/java-appsync-sam/src/test/java/com/example/MockTest.java:21]: Move long JSON strings to a file and read the file using ObjectMapper. Or prefer using multi-line strings with Java 17.

**Review by dmahapatro — CHANGES_REQUESTED** [review]

File: java-test-samples/java-appsync-sam/pom.xml [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.example</groupId>
+    <artifactId>java-appsync-sam</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
```
[the remark this answers is not in the document]
Comment (khairulhabibataws) [reply java-test-samples/java-appsync-sam/pom.xml]: done

File: java-test-samples/java-appsync-sam/README.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
+# java-appsync-sam
```
[the remark this answers is not in the document]
Comment (khairulhabibataws) [reply java-test-samples/java-appsync-sam/README.md:1]: done

File: java-test-samples/java-appsync-sam/src/test/java/com/example/MockTest.java (hunk 1 above)
Comment (khairulhabibataws) [reply java-test-samples/java-appsync-sam/src/test/java/com/example/MockTest.java]: done

**Review by muraria — APPROVED** [review]

File: java-test-samples/java-appsync-sam/README-Cognito.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Shared Resources
```
Comment (dancfox) [inline java-test-samples/java-appsync-sam/README-Cognito.md:1]: I recommend adding a paragraph to describe the purpose of the app and what it does. Would help to provide more context. Perhaps start with an **Overview** section?

File: java-test-samples/java-appsync-sam/README.md [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
```
Comment (dancfox) [inline java-test-samples/java-appsync-sam/README.md:1]: I recommend adding a paragraph to describe the purpose of the app and what it does. Would help to provide more context. Perhaps start with an Overview section?

Comment (dancfox) [reply java-test-samples/java-appsync-sam/README.md:1]: Or maybe it's a section that describes the **System Under Test**?

**Comment by dmahapatro** [comment]
@dancfox Latest commits from @khairulhabibataws should address your comments. Please feel free to merge the PR if everything looks good.

**Review by dancfox — APPROVED** [review]

**Pull request merged**